azurerm/internal/services/databricks/databricks_workspace_resource_test.go

```diff
@@ -59,15 +85,15 @@ func TestAccDatabricksWorkspace_complete(t *testing.T) {
 			Config: r.complete(data),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
-				check.That(data.ResourceName).Key("managed_resource_group_id").Exists(),
-				check.That(data.ResourceName).Key("managed_resource_group_name").Exists(),
-				check.That(data.ResourceName).Key("custom_parameters.0.virtual_network_id").Exists(),
-				check.That(data.ResourceName).Key("tags.%").HasValue("2"),
-				check.That(data.ResourceName).Key("tags.Environment").HasValue("Production"),
-				check.That(data.ResourceName).Key("tags.Pricing").HasValue("Standard"),
 			),
 		},
 		data.ImportStep(),
+		{
+			Config: r.cleanupSubnetDelegation(data),
+		},
+		{
+			Config: r.cleanupNsgAssociation(data),
+		},
```

Comment:
we should not be doing a multi step cleanup.. how come this is required?

Reply:
This is required because when you configure a Databricks resource to with `no_public_ip` azure proper will automatically create and attach a network policy to your subnets. The subnets cannot be deleted if there is a network policy attached to them, plus the subnets have a service delegation assigned to them as well which means we cannot delete the subnets or the VNet until the network policy and the service delegation association with the Databricks workspace has been broken.